File: 2395. Find Subarrays With Equal Sum/Solution.cpp
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
    bool findSubarrays(vector<int>& nums)
    {
        unordered_set<int> seen;
        for (size_t i = 1; i < nums.size(); i++)
            if (!seen.insert(nums[i] + nums[i - 1]).second)
                return true;
        return false;
    }
};

File: 2395. Find Subarrays With Equal Sum/Solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean findSubarrays(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int i = 1; i < nums.length; i++) {
            if (!set.add(nums[i - 1] + nums[i])) {
                return true;
            }
        }
        return false;
    }
}
